### PR TITLE
Improve CI-CD workflows

### PR DESCRIPTION
**Pull request by isuruh15**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: .github/workflows/cd-executor.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: Publish Release
+
+on:
+    workflow_call:
+      inputs:
+        target-branch:
```
Comment (ThishaniLucas) [inline .github/workflows/cd-executor.yml:6]: We don't need the branch as an input. When triggering the workflow itself, there's an option to select the branch.

Comment (isuruh15) [reply .github/workflows/cd-executor.yml]: fixed in [d4ede32](https://github.com/wso2/open-healthcare-prebuilt-services/pull/45/commits/d4ede32dcac99e3d755eeaddbfc9987a4c049cb2)

File: .github/workflows/cd-hl7v2-to-fhir.yml [hunk 2]
```diff
@@ -8,5 +8,4 @@ jobs:
      uses: ./.github/workflows/cd-executor.yml
      secrets: inherit
      with:
-      working_dir: ./transformations/v2-to-fhirr4-service
-      service-name: hl7v2-to-fhir-transformation
\ No newline at end of file
+      working_dir: ./transformations/v2-to-fhirr4-service
```
Comment (ThishaniLucas) [inline .github/workflows/cd-hl7v2-to-fhir.yml:11]: workflow has defined input as working-dir, but here working_dir is used.

File: .github/workflows/cd-executor.yml [hunk 3]
```diff
@@ -1,60 +1,115 @@
...
-       run: |
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Run ballerina build
+        uses: ballerina-platform/ballerina-action@2201.8.4
+        with:
+          args: pack
+        env:
+          working-dir: ${{ inputs.working-dir }}
```
Comment (ThishaniLucas) [inline .github/workflows/cd-executor.yml:24]: for bal build we need to use WORKING_DIR

Comment (isuruh15) [reply .github/workflows/cd-executor.yml]: fixed with [6d9bf5b](https://github.com/wso2/open-healthcare-prebuilt-services/pull/45/commits/6d9bf5b2891bd6c99c43d27980ec381215719a9d)

File: .github/workflows/cd-hl7v2-to-fhir.yml (hunk 2 above)
Comment (isuruh15) [reply .github/workflows/cd-hl7v2-to-fhir.yml:11]: fixed with [6d9bf5b](https://github.com/wso2/open-healthcare-prebuilt-services/pull/45/commits/6d9bf5b2891bd6c99c43d27980ec381215719a9d)

**Pull request merged**